src/services/Branches.js

```diff
@@ -16,14 +16,10 @@ class Branches extends BaseService {
     });
   }
 
-  protect(projectId, branchName, options) {
-    const [pId, bName] = [projectId, branchName].map(encodeURIComponent);
+  protect(projectId, options) {
+    const pId = encodeURIComponent(projectId);
```

Comment:
Based on this: https://docs.gitlab.com/ee/api/protected_branches.html#protect-repository-branches, the name of the branch is still mandatory. So instead of just having the generic options could it be more like:

```
 protect(projectId, branchName, options) {
     const pId = encodeURIComponent(projectId); 

     return RequestHelper.post(this, `projects/${pId}/protected_branches`, {
         name: branchName, 
         ...options
      });
}
```

Reply:
sure, I can update it.